internal/pkg/service/stream/storage/level/local/diskwriter/network/router/assignment/assign_test.go

```diff
@@ -0,0 +1,196 @@
+			Volumes:     []volume.ID{"volume1", "volume2", "volume3"},
+			ExpectedAssignment: map[string][]volume.ID{
+				"source1": {"volume1", "volume2", "volume3"},
+			},
+		},
+		{
+			Name:        "2 sources, 1 volume",
+			SourceNodes: []string{"source1", "source2"},
+			Volumes:     []volume.ID{"volume1"},
+			ExpectedAssignment: map[string][]volume.ID{
+				"source1": {"volume1"},
+				"source2": {"volume1"},
+			},
+		},
+		{
+			Name:        "2 sources, 2 volumes",
+			SourceNodes: []string{"source1", "source2"},
+			Volumes:     []volume.ID{"volume1", "volume2"},
+			ExpectedAssignment: map[string][]volume.ID{
+				"source1": {"volume1"},
+				"source2": {"volume2"},
+			},
+		},
+		{
+			Name:        "3 sources, 1 volume",
+			SourceNodes: []string{"source1", "source2", "source3"},
+			Volumes:     []volume.ID{"volume1"},
+			ExpectedAssignment: map[string][]volume.ID{
+				"source1": {"volume1"},
+				"source2": {"volume1"},
+				"source3": {"volume1"},
+			},
+		},
+		{
+			Name:        "3 sources, 2 volumes",
+			SourceNodes: []string{"source1", "source2", "source3"},
+			Volumes:     []volume.ID{"volume1", "volume2", "volume3"},
+			ExpectedAssignment: map[string][]volume.ID{
+				"source1": {"volume1"},
+				"source2": {"volume2"},
+				"source3": {"volume3"},
+			},
+		},
+		{
+			Name:        "3 sources, 3 volumes",
+			SourceNodes: []string{"source1", "source2", "source3"},
+			Volumes:     []volume.ID{"volume1", "volume2", "volume3"},
+			ExpectedAssignment: map[string][]volume.ID{
+				"source1": {"volume1"},
+				"source2": {"volume2"},
+				"source3": {"volume3"},
+			},
+		},
+		{
+			Name:             "3 sources, 3 volumes, min 2 per node",
+			SourceNodes:      []string{"source1", "source2", "source3"},
+			Volumes:          []volume.ID{"volume1", "volume2", "volume3"},
+			MinSlicesPerNode: 2,
+			ExpectedAssignment: map[string][]volume.ID{
+				"source1": {"volume1", "volume2"},
+				"source2": {"volume3", "volume1"},
+				"source3": {"volume2", "volume3"},
+			},
+		},
+		{
+			Name:        "8 sources, 3 volumes",
+			SourceNodes: []string{"source1", "source2", "source3", "source4", "source5", "source6", "source7", "source8"},
+			Volumes:     []volume.ID{"volume1", "volume2", "volume3"},
+			ExpectedAssignment: map[string][]volume.ID{
+				"source1": {"volume1"},
+				"source2": {"volume2"},
+				"source3": {"volume3"},
+				"source4": {"volume1"},
+				"source5": {"volume2"},
+				"source6": {"volume3"},
+				"source7": {"volume1"},
+				"source8": {"volume2"},
+			},
+		},
+		{
+			Name:        "3 sources, 8 volumes",
+			SourceNodes: []string{"source1", "source2", "source3"},
+			Volumes:     []volume.ID{"volume1", "volume2", "volume3", "volume4", "volume5", "volume6", "volume7", "volume8"},
+			ExpectedAssignment: map[string][]volume.ID{
+				"source1": {"volume1", "volume2", "volume3"},
+				"source2": {"volume4", "volume5", "volume6"},
+				"source3": {"volume7", "volume8", "volume1"},
+			},
```

Comment:
Example of assignment of volumes/slices to source nodes:

We need some redundancy, but we don't want all source nodes to write to all volumes.